Sources/Core/GTMSessionUploadFetcher.m

```diff
@@ -1755,7 +1760,7 @@ - (int64_t)updateChunkFetcher:(GTMSessionFetcher *)chunkFetcher forChunkAtOffset
   GTMSESSION_ASSERT_DEBUG(offset < fullUploadLength || fullUploadLength == 0,
                           @"offset %lld exceeds data length %lld", offset, fullUploadLength);
 
-  if (granularity > 0) {
+  if (granularity > 0 && offset != fullUploadLength) {
```

Comment:
offset < fullUploadLength

? … just for sanity checking?

Reply:
Done.